freqtrade/freqtradebot.py

```diff
@@ -260,19 +280,14 @@ def create_trade(self) -> bool:
         :param interval: Ticker interval used for Analyze
         :return: True if a trade object has been created and persisted, False otherwise
         """
-        stake_amount = self.config['stake_amount']
         interval = self.analyze.get_ticker_interval()
+        stake_amount = self._get_trade_stake_amount()
```

Comment:
What happens if `stake_amount == 0.0`?
I think we should return from `create_trade()` in this case (additionally with some logging), because as far as I can see there are a lot of unneeded exchange calls which can be skipped.

Reply:
Nice catch
should there also be a lower limit of available stake-currency where it doesn't create new trades?
say available stake is 0.0001 ETH ... it's not 0 - but still should not create another trade as it would hit the lower limit for most exchanges (at least binance).

Reply:
An exchange API returns the minimum possible amount for trading and it depends on pair. So I think I will add this check as well.

Reply:
if you buy "minimum possible" and don't have BNB (at least for binance) - you'll not be able to sell it afterwards, as the fee is subtracted from what you buy.

i would make it configurable as `minimum_stake` ... which only applies if `unlimited` is set.

Reply:
I would go with the minimum possible amount returned by the exchange plus some small amount, lets say: `minimum_stake *= 1.05` to cover cases such as BNB (still not great, but it should cover most cases).
I'm not sure if making it configurable adds value, because it can be very different from coin to coin.

Reply:
@gcarq I have the same thoughts with a small correction:
`min_required_amount = minimum_pair_stake * (1 - stoploss + 0.02)`
It will cover the case when stoploss is reached and a bot will sell a coin.

Just not sure what constant to choose 2%, 3% or 5%. What do you think?

I think `minimum_stake` is useless, because in most cases user doesn't know the real `minimum_pair_stake` and can set `minimum_stake < minimum_pair_stake`. Also `minimum_pair_stake` can depend on the order type `buy`/`sell`.

Reply:
@arudov Good idea to calculate the stoploss in. I would go with 5% buffer in the start and see how it behaves, but I'm also not sure.

Reply:
@gcarq I've tried to implement that logic and then I realized that it requires addition API calls. This calls should get the information about pairs (CCXT fetch_markets call). This information could be cached, but it should be updated with some time period. Then I thought what will happen if we had this code: the bot would generate an exception "Not enough stake amount". But if we do not add this check the bot will fail on "create_trade" call and the exchange will return a corresponding error code, explaining that a trade of such size could't be created.

So I do not see any good reasons to add an extra API calls to the current code. I've added the check that stake_amount is more than 0.0005 as it is for static stake_amount.

Reply:
The markets-information is available cached as `exchange._API.markets` (not sure about the s at the end) ... it's initialized when `_API.load_markets()` is called - which needs to happen for almost every call to ccxt anyway.
We could fairly easily expose that dict through a proper interface in the exchange class if necessary as i needed this in the download-script too.

I agree that this would need some reloading at some point or another - will also need to check if ccxt handles caching / reloading under the hood already for us.

I'm honestly not sure about hardcoding 0.0005 - for BTC this will (probably) be fine, but what about ETH or USDT?

Reply:
Yes my point is that anywhere the "create order" call fails if the order size is too small. So there is no difference if the exception is generated here, during calculating the order size, or few lines later during "create order" call. Anywhere the bot will stop trading.

`0.0005` was taken from here: https://github.com/freqtrade/freqtrade/blob/bd6ed3ada4fadb01641f0598fcd838a776d7ce75/freqtrade/constants.py#L47

Reply:
I also think that we should expose the _API.markets and get the minimum amount from there instead of hardcoding it. The `0.0005` is outdated already since we're able to trade with coins and exchanges other than BTC and bittrex. I know that the exceptions in both cases will be catched, but I think we should prefer the clean way